### PR TITLE
Feature/cdap 1666 use id program instead of also programtype

### PR DESCRIPTION
**Pull request by anwar6953**
Simply refactor method signatures to resolve the following JIRA.

https://issues.cask.co/browse/CDAP-1666
http://builds.cask.co/browse/CDAP-DUT1938-4

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java [hunk 1]
```diff
@@ -386,174 +369,4 @@ protected boolean respondIfElementNotFound(Throwable t, HttpResponder responder)
     }
     return false;
   }
-
-  // TODO: refactor
-  protected final void dataList(HttpResponder responder, Store store, DatasetFramework dsFramework,
```
Comment (henry-cask) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java]: Just curious why this block of code dissapear? Not used anywhere else?

Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java]: Correct. It was used before, but then this functionality is replaced by the Datasets Registry concept.
You can see usages of it being removed here, but the method itself was not removed.
https://github.com/caskdata/cdap/pull/2166/files#diff-2b14ed88e3eb39a9c4550dfd17f60b8aL80

**Comment by henry-cask** [comment]
LGTM

+1

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java [hunk 2]
```diff
@@ -328,18 +325,16 @@ void addApplication(Id.Application id,
   /**
    * Deletes a schedules from a particular program
    * @param program defines program from which a schedule is being deleted
-   * @param programType defines the type of the program
    * @param scheduleName the name of the schedule to be removed from the program
    */
-  void deleteSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName);
```
Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java]: question: why do we need SchedulableProgramType ? i noticed the programType includes all of the types in SchedulableProgramType

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/program/ProgramBundle.java [hunk 3]
```diff
@@ -57,15 +56,13 @@ public boolean apply(JarEntry input) {
...
+  public static Location create(Id.Program programId, ArchiveBundler bundler, Location output,
+                                String className, ApplicationSpecification appSpec) throws IOException {
+    return create(programId, bundler, output, className, appSpec, null);
   }
 
-  public static Location create(Id.Application id, ArchiveBundler bundler, Location output, String programName,
-                               String className, ProgramType type, ApplicationSpecification appSpec,
-                               Manifest other) throws IOException {
+  public static Location create(Id.Program programId, ArchiveBundler bundler, Location output,
+                               String className, ApplicationSpecification appSpec, Manifest other) throws IOException {
```
Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/program/ProgramBundle.java]: add nullable for manifest ?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java [hunk 4]
```diff
@@ -245,8 +245,8 @@ private void deleteProgramLocations(Id.Application appId) throws IOException {
     try {
       Id.Program programId = Id.Program.from(appId.getNamespaceId(), appId.getId(),
                                              ProgramType.WEBAPP, ProgramType.WEBAPP.name().toLowerCase());
-      Location location = Programs.programLocation(namespacedLocationFactory, appFabricDir, programId,
-                                                   ProgramType.WEBAPP);
+      Location location = Programs.programLocation(namespacedLocationFactory, appFabricDir, programId
+      );
```
Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java]: formatting: move this to previous line

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java (hunk 2 above)
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java]: Not all `ProgramType`s can be passed to scheduler APIs, and by `SchedulableProgramType` being a subset of `ProgramType`, this helps restrict which `ProgramType`s are allowed to be scheduled/unscheduled.

Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java]: okay, just wanted to clarify.

**Comment by gsps1** [comment]
LGTM :+1:

**Pull request merged**